tests/flamegraph.rs

```diff
@@ -346,3 +345,169 @@ fn flamegraph_grey_frames() {
+}
+
+#[test]
+fn flamegraph_notes_with_symbols() {
+    let input_file = "./tests/data/differential/perf-cycles-instructions-01-collapsed-all-diff.txt";
+    let expected_result_file = "./tests/data/options/notes_with_symbols.svg";
+
+    let options = flamegraph::Options {
+        notes: "Test Notes <& ' \"".to_owned(),
+        ..Default::default()
+    };
+
+    test_flamegraph(input_file, expected_result_file, options).unwrap();
+}
+
+#[test]
+fn flamegraph_count_name_simple() {
+    let input_file = "./tests/data/differential/perf-cycles-instructions-01-collapsed-all-diff.txt";
+    let expected_result_file = "./tests/data/options/count_name_simple.svg";
+
+    let options = flamegraph::Options {
+        count_name: "test-samples".to_owned(),
+        ..Default::default()
+    };
+
+    test_flamegraph(input_file, expected_result_file, options).unwrap();
+}
+
+#[test]
+fn flamegraph_count_name_with_symbols() {
+    let input_file = "./tests/data/differential/perf-cycles-instructions-01-collapsed-all-diff.txt";
+    let expected_result_file = "./tests/data/options/count_name_with_symbols.svg";
+
+    let options = flamegraph::Options {
+        count_name: "test-samples <& ' \"".to_owned(),
+        ..Default::default()
+    };
+
+    test_flamegraph(input_file, expected_result_file, options).unwrap();
+}
+
+#[test]
+fn flamegraph_name_type_simple() {
+    let input_file = "./tests/data/differential/perf-cycles-instructions-01-collapsed-all-diff.txt";
+    let expected_result_file = "./tests/data/options/name_type_simple.svg";
+
+    let options = flamegraph::Options {
+        name_type: "Tfunction:".to_owned(),
+        ..Default::default()
+    };
+
+    test_flamegraph(input_file, expected_result_file, options).unwrap();
+}
+
+#[test]
+fn flamegraph_name_type_with_quote() {
+    let input_file = "./tests/data/differential/perf-cycles-instructions-01-collapsed-all-diff.txt";
+    let expected_result_file = "./tests/data/options/name_type_with_quote.svg";
+
+    let options = flamegraph::Options {
+        name_type: "Test: '".to_owned(),
+        ..Default::default()
+    };
+
+    test_flamegraph(input_file, expected_result_file, options).unwrap();
+}
+
+#[test]
+fn flamegraph_font_type_simple() {
+    let input_file = "./tests/data/differential/perf-cycles-instructions-01-collapsed-all-diff.txt";
+    let expected_result_file = "./tests/data/options/font_type_simple.svg";
+
+    let options = flamegraph::Options {
+        font_type: "Andale Mono".to_owned(),
+        ..Default::default()
+    };
+
+    test_flamegraph(input_file, expected_result_file, options).unwrap();
+}
+
+#[test]
+fn flamegraph_font_type_with_quote() {
+    let input_file = "./tests/data/differential/perf-cycles-instructions-01-collapsed-all-diff.txt";
+    let expected_result_file = "./tests/data/options/font_type_with_quote.svg";
+
+    let options = flamegraph::Options {
+        font_type: "Andale Mono\"".to_owned(),
```

Comment:
I think you want this to contain a single quote `'`, not a double quote `"`.

Reply:
Font type  is used inside the xml tags:
```<text text-anchor="left" x="13.00" y="159.50" font-size="12" font-family="Verdana" fill="rgb(0, 0, 0)">__libc_start_main</text>```

So only it makes sense to check `"` for escaping.

Reply:
Oh, yes, sorry, I got confused with `opt.name_type` that we're discussing above :)